packages/discovery-provider/src/tasks/entity_manager/entities/playlist.py

```diff
@@ -110,6 +111,57 @@ def update_playlist_routes_table(
     )
 
 
+def update_playlist_tracks_relations(
+    params: ManageEntityParameters, playlist_record: Playlist
+):
+    # Update the playlist_tracks_relations table
+    session = params.session
+    existing_playlist_tracks_relations = (
+        session.query(CollectionTrackRelation)
+        .filter(
+            CollectionTrackRelation.collection_id == params.entity_id,
+        )
+        .all()
+    )
+    existing_tracks = {
+        track.track_id: track for track in existing_playlist_tracks_relations
+    }
+    playlist = helpers.model_to_dictionary(playlist_record)
+    updated_track_ids = [
+        track["track"] for track in playlist["playlist_contents"]["track_ids"]
+    ]
+    params.logger.info(
+        f"playlists.py | Updating playlist tracks relations for {playlist['playlist_id']}"
+    )
+
+    # delete relations that previously existed but are not in the updated list
+    for relation in existing_playlist_tracks_relations:
+        if relation.track_id not in updated_track_ids:
+            relation.is_delete = True
+            relation.updated_at = params.block_datetime
+
+    # add row for each track that is not already in the table
+    for track_id in updated_track_ids:
+        if track_id not in existing_tracks:
+            new_collection_track_relation = CollectionTrackRelation(
+                collection_id=playlist["playlist_id"],
+                track_id=track_id,
+                is_delete=False,
+                created_at=params.block_datetime,
+                updated_at=params.block_datetime,
+            )
+            # upsert to handle duplicates
```

Comment:
is the upsert in this case to prevent situation where updated_track_ids might have the same ids twice?